tests/lib/config-array-factory.js

```diff
@@ -2066,8 +2235,9 @@ describe("ConfigArrayFactory", () => {
                 });
             });
 
-            it("should load two separate configs from a plugin", () => {
-                const { ConfigArrayFactory } = defineConfigArrayFactoryWithInMemoryFileSystem({
+            it("should load two separate configs from a plugin", async () => {
```

Comment:
This seems to be a problem with the require cache.

We're using [require](https://github.com/eslint/eslintrc/blob/b347ca00f93bfb12144ef1e6cea25607e07a5b3c/lib/config-array-factory.js#L1030) to load plugins, and since this one has the same path as in the previous test, it will return the cached version.

Renaming to `eslint-plugin-test1` would fix the problem. Maybe we could also think about deleting require cache for all modules in the temp dir before each test, as this might be happening with other tests but went unnoticed.

Reply:
Ah, brilliant!